javascript/react/src/components/ArqueroDetailsList/ArqueroDetailsList.hooks.ts

```diff
@@ -169,16 +169,19 @@ export function useListProps(version: number): IListProps<any> {
 }
 
 export function useFields(table: ColumnTable, fields?: Field[]): Field[] {
-	// TODO: this uses the older type determination, which is not as sophisticated as our new codebook.
-	// however, dates are well detected in the new codebooks, but not well parsed.
+	// TODO: this uses a bit of a kludge, because codebooks are not updated when the workflow produces new columns
+	// this supplements the known fields with guesses from the table
 	return useMemo(() => {
-		if (fields) {
-			return fields
-		}
 		const types = columnTypes(table)
-		return Object.entries(types).map(([name, type]) => ({
-			name,
-			type,
-		}))
+		return Object.entries(types).map(([name, type]) => {
+			const existing = fields?.find((f) => f.name === name)
+			return (
+				existing || {
```

Comment:
existing ??

Reply:
This is an old habit from before ?? existed. Are you aware of any lint rules that might catch these? I don't see anything on eslint other than support for the syntax.

Reply:
Not that I'm aware of - it's a pretty subtle difference between the two operators

Reply:
It is - though I suspect all of our assignments that use || could use ??